### PR TITLE
fix(gs): Set lastUnpausedDirection for newly created players

### PR DESCRIPTION
**Pull request by abhishek276533**
Fixes: #25

### CODE REVIEW

**Comment by jespertheend** [comment]
Thanks for the contribution!
It seems like I'm still able to reproduce this. Not sure why though.
I think another fix might be to set `#lastUnpausedDirection` in the constructor as well, as opposed to only `#currentDirection`.

**Comment by abhishek276533** [comment]
> Thanks for the contribution! It seems like I'm still able to reproduce this. Not sure why though. I think another fix might be to set `#lastUnpausedDirection` in the constructor as well, as opposed to only `#currentDirection`.

You're welcome! I'm glad I could contribute. I've added the change, Let me know if you still face any issue

File: gameServer/src/gameplay/Player.js [hunk 1]
```diff
@@ -174,9 +174,11 @@ export class Player {
 		const { position, direction } = game.getNewSpawnPosition();
 		this.#currentPosition = position;
 		this.#currentDirection = direction;
+		this.#lastUnpausedDirection = direction === "paused" ? this.#lastUnpausedDirection : direction;
 		this.#lastEdgeChunkSendX = this.#currentPosition.x;
 		this.#lastEdgeChunkSendY = this.#currentPosition.y;
 		this.#currentPositionChanged();
+		this.clientPosUpdateRequested(direction, position);
```
Comment (jespertheend) [inline gameServer/src/gameplay/Player.js:181]: I think this line is no longer necessary?

File: gameServer/src/gameplay/Player.js [hunk 2]
```diff
@@ -174,9 +174,11 @@ export class Player {
 		const { position, direction } = game.getNewSpawnPosition();
 		this.#currentPosition = position;
 		this.#currentDirection = direction;
+		this.#lastUnpausedDirection = direction === "paused" ? this.#lastUnpausedDirection : direction;
```
Comment (jespertheend) [inline gameServer/src/gameplay/Player.js:177]: `getNewSpawnPosition()` guaranteed to never return "paused" as direction.
So you can use `this.#lastUnpausedDirection = direction;` instead.
To get rid of the type error you can add a new `@type {Exclude<Direction, "paused">} UnpausedDirection` type right below the `Direction` type in Player.js.
You can then replace the two `import("./Player.js").Direction` references from `getNewSpawnPosition()`.
Finally the type of `Player.#lastUnpausedDirection` can also be replaced with that new type.

Let me know if that makes sense.

File: gameServer/src/gameplay/Player.js (hunk 1 above)
Comment (abhishek276533) [reply gameServer/src/gameplay/Player.js]: Yes, we can remove it. have removed in latest commit

File: gameServer/src/gameplay/Player.js (hunk 2 above)
Comment (abhishek276533) [reply gameServer/src/gameplay/Player.js]: Sure, we can update it that way. Have added the changes in the latest commit.

**Review by jespertheend — APPROVED** [review]
Thanks LGTM!

**Comment by jespertheend** [comment]
Oops, wait. Seems like ci is failing. You can run `deno fmt` to format the files.
Make sure to `cd` into the project folder first or it will, somewhat annoyingly, start formatting all files on your hard disk.

**Comment by jespertheend** [comment]
Thanks!

**Pull request merged**